hypershift-operator/controllers/hostedcluster/internal/platform/agent/agent.go

```diff
@@ -45,13 +53,129 @@ func (p Agent) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, create
 }
 
 func (p Agent) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, tokenMinterImage string) (*appsv1.DeploymentSpec, error) {
-	// It is expected that cluster-api-provider-agent and the Infrastructure Operator are installed as a prerequisite
-	return nil, nil
+	deploymentSpec := &appsv1.DeploymentSpec{
+		Replicas: k8sutilspointer.Int32Ptr(1),
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"control-plane": "controller-manager",
```

Comment:
Labels are set consistently across platforms here https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L1183-L1196 so no need to set any additional here

Reply:
Done